assignment1/KthToLastElement_Test.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+
+from linked_list import LinkedList, kth_to_last_element
+import unittest
+
+class KthToLastElement_Test(unittest.TestCase):
+
+    def setUp(self):
+        self.__linked_list = LinkedList()
+        self.__linked_list.add_first(0)
+        self.__linked_list.add_first(1)
+        self.__linked_list.add_first(2)
+        self.__linked_list.add_first(3)
+        self.__linked_list.add_first(4)
+        self.__linked_list.add_first(5)
+        self.__linked_list.add_first(6)
+        self.__linked_list.add_first(7)
+        self.__linked_list.add_first(8)
+        self.__list_head = self.__linked_list.get_head()
+        
+    def testBaseCases(self):
+        self.__max_value = self.__linked_list.size()
+        for i in range(0, self.__max_value):
+            self.assertEqual(kth_to_last_element(self.__list_head, i), i)
+        
+    def testErrorCases(self):
+        with self.assertRaises(ValueError): 
```

Comment:
The way the `with`-statement works is that it calls the `__exit__()` method on the object it created (the result of the call to `assertRaises()` in this case)  after executing the code within the `with`-statement, even if there is an exception. Basically it is similar to a `try/finally`-statement. This is used in this case to check that the assertion is actually raised. Compare with:

```python
  try:
    run()
    self.assertFalse("should not happen")
  except Exception, e:
    pass  # This is expected.
```

In Java, the way one can test for an exception can either be with a `try/finally`-statement (similar to the above); with the annotation `@Test(expected = XXX.class)`; or with a rule like `@Rule public final ExpectedException exception = ExpectedException.none();`.